lib/typecheck/nullable_env.ml

```diff
@@ -5,18 +5,18 @@
  *)
 type t = Ty_env.t option
 
-let intersect nenv1 nenv2 =
+let intersect nenv1 nenv2 pos =
```

Comment:
Probably better as optional

Reply:
If I changed this, the `pos` argument in join/combine/intersect in `ty_env.ml` should also as optional right?